regression/esbmc/01_pthread48/main.c

```diff
@@ -0,0 +1,53 @@
+/*
+ * Extracted from https://github.com/sosy-lab/sv-benchmarks.
+ * Contributed by Vladimír Štill (https://divine.fi.muni.cz).
+ * Description: A test case for pthread TLS.
+ */
+
+#include <errno.h>
+#include <pthread.h>
+#include <assert.h>
+
+void *worker(void *k)
+{
+  pthread_key_t *key = k;
+
+  long val = (long)pthread_getspecific(*key);
+  assert(val == 0);
+
+  int r = pthread_setspecific(*key, (void *)42);
+  assert(r == 0 || r == ENOMEM);
+
+  val = (long)pthread_getspecific(*key);
+  assert(val == 42);
+
+  return 0;
+}
+
+int main()
+{
+  pthread_key_t key;
+  int r = pthread_key_create(&key, NULL);
+  assert(r == 0 || r == EAGAIN || r == ENOMEM);
```

Comment:
This assertion looks dodgy: we either succeeded with 0 and the program should continue, or we failed and all set/getspecific will return wrong values.

This program is testing that it doesn' t matter if we failed or succeeded, the key will always be there.

Reply:
Fixed.